src/umpire/op/CMakeLists.txt

```diff
@@ -67,7 +67,7 @@ if (ENABLE_CUDA)
 
   set (umpire_op_depends
     ${umpire_op_depends}
-    cuda_runtime)
+    cuda)
```

Comment:
I'm not sure this is required? The cuda dep (required by the DeviceAllocator) should be handled by the code added in the src/umpire/CMakeLists.txt

Reply:
Hi @davidbeckingsale, I just commented this out and gave it a try. I am still getting errors like this:

```
Umpire/src/umpire/op/CudaCopyFromOperation.cpp:46:42: error: ‘Cuda’ is not a member of ‘camp::resources’
   auto device = ctx.get<camp::resources::Cuda>();
```

Again,  I think this is more of an issue with the `#ifdef __CUDA_ARCH__` guards in `camp`, which would only work if the code is fed into the NVCC compiler.

Also, note that the code in `src/umpire/CMakeLists.txt` adds the `cuda` dep to `umpire_depends`, which, is not used in this file. This file uses the `umpire_op_depends`.  

Instead of adding the `cuda` dep here, would you rather add `umpire_depends` to it? (I think that would work as well.)

Thanks again for taking a look. Let me know if you'd like me to make any changes or try anything else.

Reply:
I took bit more of a look into this, and we are handling the camp/cuda stuff here: https://github.com/LLNL/Umpire/blob/develop/src/umpire/tpl/CMakeLists.txt#L89-L92 Would a similar solution work for your project?

I agree we should have the cuda dep for the main library.

Reply:
Thanks for looking into this. Yes, adding the definition `CAMP_HAVE_CUDA` to `camp` when building with CUDA works for me. I'll handle that on my end.

Reply:
Great, I'll remove this change from #581 and then get this merged.

Reply:
Fantastic -- Thanks!

Reply:
Thank you for finding and fixing this!